usecases/ratelimiter/limiter_test.go

```diff
@@ -102,3 +102,11 @@ func TestLimiterCantGoNegative(t *testing.T) {
 	}
 	assert.False(t, l.TryInc())
 }
+
+func BenchmarkLimiter(b *testing.B) {
+	l := New(-1)
```

Comment:
Negative max limit short-circuits TryInc and Dec so this just benchmarks integer comparison, moreover proper benchmark should test goroutine contention.

Reply:
IMO previous implementation was correct and should not be changed unless proven to be a bottleneck.